### PR TITLE
Initial port to Swift 4

### PR DESCRIPTION
**Pull request by tcldr**
What's in this pull request?
============================

Initial migration of library to Swift 4.

Why merge this pull request?
============================

Migrates library to latest version of Swift. All tests passing with the Swift 4 version of SwiftCheck that I submitted as a PR earlier.

What's worth discussing about this pull request?
================================================

This PR focuses on getting the build compiling without warning and passing all tests. I will have missed anything not picked up by these! There may be some wider changes you have in mind to take advantage of Swift 4. 

What downsides are there to merging this pull request?
======================================================

Uses temporary refs to my personal SwiftCheck/Swiftx forks with Swift 4 migration. Will need updating to point to official SwiftCheck/Swiftx repos.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +135,13 @@ To add Swiftz to your application:
 - Set the directory to `Frameworks`
 - Click the + and add Swiftz
  
+**Using Swift Package Manager**
+
+- Add Swiftz to your `Package.swift` file's `dependencies` section:
+
+```swift
+.Package(url: "https://github.com/typelift/Swiftz", versions: Version(0,6,0)..<Version(1,0,0))
```
Comment (CodaFi) [inline README.md]: This is using the old-style package manifest definition.  Is that intentional?

Comment (tcldr) [reply README.md]: Strange. I don't remember adding that!

**Comment by tcldr** [comment]
FYI This still points to my fork of Swiftx as some files were moved around to work with the SPMv4 format.

**Comment by CodaFi** [comment]
Meh, I'll take it as-is.

**Pull request merged**